packages/clerk-js/src/core/resources/BillingCheckout.ts

```diff
@@ -91,3 +99,162 @@ export class BillingCheckout extends BaseResource implements BillingCheckoutReso
     );
   };
 }
+
+export const createSignals = () => {
+  const resourceSignal = signal<{ resource: CheckoutFuture | null }>({ resource: null });
+  const errorSignal = signal<{ error: ClerkError | null }>({ error: null });
+  const fetchSignal = signal<{ status: 'idle' | 'fetching' }>({ status: 'idle' });
+  const computedSignal = computed<Omit<CheckoutSignalValue, 'checkout'> & { checkout: CheckoutFutureResource | null }>(
+    () => {
+      const resource = resourceSignal().resource;
+      const error = errorSignal().error;
+      const fetchStatus = fetchSignal().status;
+
+      const errors = errorsToParsedErrors(error);
+      return { errors: errors, fetchStatus, checkout: resource };
+    },
+  );
+
+  return { resourceSignal, errorSignal, fetchSignal, computedSignal };
+};
+
+export class CheckoutFuture implements CheckoutFutureResourceLax {
```

Comment:
We made `SignInFuture` because we needed to keep the API from `SignIn` around for backwards compatibility. Do we need to maintain that for Core 3? Or can we just make this API the default API, and move all these functions to the base `BillingCheckout` class?